tests/monitors/statsd/statsd_test.py

```diff
@@ -19,57 +23,72 @@ def test_statsd_monitor():
         assert wait_for(
             p(has_datapoint_with_metric_name, backend, "gauge.statsd.test")
         ), "Didn't get statsd.test metric"
-        assert wait_for(p(has_datapoint_with_dim, backend, "foo", "bar")), "Didn't get foo dimension"
 
 
-@pytest.mark.k8s
-@pytest.mark.kubernetes
-def test_statsd_in_k8s(agent_image, minikube, k8s_test_timeout, k8s_namespace):
-    # hack to populate data for statsd
-    minikube.container.exec_run(
-        [
-            "/bin/bash",
-            "-c",
-            'n=0; while [ $n -le %d ]; do \
-            echo "statsd.[foo=bar,dim=val]test:1|g" | nc -w 1 -u 127.0.0.1 8125; \
-            sleep 1; \
-            (( n += 1 )); \
-            done'
-            % k8s_test_timeout,
-        ],
-        detach=True,
-    )
-    monitors = [
-        {
-            "type": "collectd/statsd",
-            "listenAddress": "127.0.0.1",
-            "listenPort": 8125,
-            "counterSum": True,
-            "deleteSets": True,
-            "deleteCounters": True,
-            "deleteTimers": True,
-            "deleteGauges": True,
-        }
-    ]
-    run_k8s_monitors_test(
-        agent_image,
-        minikube,
-        monitors,
-        namespace=k8s_namespace,
-        expected_metrics={"gauge.statsd.test"},
-        expected_dims={"foo", "dim"},
-        test_timeout=k8s_test_timeout,
-    )
+def test_statsd_monitor_prefix():
+    """
+    Test metric prefix
+    """
+    with run_agent(
+        """
+monitors:
+  - type: statsd
+    listenAddress: localhost
+    listenPort: 8126
+    metricPrefix: statsd.appmesh
+"""
+    ) as [backend, _, _]:
+        assert wait_for(p(udp_port_open_locally_netstat, 8126)), "statsd port never opened!"
+        send_udp_message(
+            "localhost",
+            8126,
+            "statsd.appmesh.cluster.cds_egress_ecommerce-demo-mesh_gateway-vn_tcp_8080.update_success:8|c",
+        )
+
+        assert wait_for(
+            p(
+                has_datapoint_with_metric_name,
+                backend,
+                "derive.cluster.cds_egress_ecommerce-demo-mesh_gateway-vn_tcp_8080.update_success",
+            )
+        ), "Didn't get statsd.test metric"
+
+
+def test_statsd_monitor_conversion():
+    """
+    Test conversion
+    """
+    with run_agent(
+        """
+monitors:
+  - type: statsd
+    listenAddress: localhost
+    listenPort: 8127
+    converters:
+    - pattern: 'cluster.cds_{traffic}_{mesh}_{service}-vn_{}.{action}'
+      metricName: '{traffic}.{action}'
+"""
+    ) as [backend, _, _]:
+        assert wait_for(p(udp_port_open_locally_netstat, 8127)), "statsd port never opened!"
+        send_udp_message(
+            "localhost", 8127, "cluster.cds_egress_ecommerce-demo-mesh_gateway-vn_tcp_8080.update_success:8|c"
+        )
+
+        assert wait_for(
+            p(has_datapoint_with_metric_name, backend, "derive.egress.update_success")
```

Comment:
Do this: 

`p(has_datapoint, metric_name="derive.egress.update_success", dimensions={"traffic": "egress", "mesh": ...})` to assert that it is getting all those things in a single datapoint.